libp2p/switch.nim

```diff
@@ -487,44 +466,11 @@ proc muxerHandler(s: Switch, muxer: Muxer) {.async, gcsafe.} =
 
   try:
     await s.identify(muxer)
-  except CatchableError as exc:
+  except IdentifyError as exc:
```

Comment:
what if the remote end closes the identify connection early and a streameof happens on the connection that identify was using?

Reply:
Good point, we should handle those as well.